src/Compilers/Core/Portable/FileSystem/FileUtilities.cs

```diff
@@ -372,6 +372,21 @@ internal static DateTime GetFileTimeStamp(string fullPath)
             }
         }
 
+        /// <exception cref="IOException"/>
+        internal static long GetFileLength(string fullPath)
+        {
+            Debug.Assert(PathUtilities.IsAbsolute(fullPath));
+            try
+            {
+                var info = new FileInfo(fullPath);
+                return info.Length;
+            }
+            catch (Exception e)
+            {
+                throw new IOException(e.Message, e);
```

Comment:
Sure.  It's only useful though if we are diligent and wrap every single place which throws an exception with IOException.  That's a whack a mole problem.  Every time we find a new place, wrap wrap warp.

Much simpler: eliminate the middle man and just catch exception.  The end result is indistinguishable.